File: apple/Sources/ReactNativeWatchOS/Modifiers/LabelReservedIconWidth.swift
import SwiftUI

/// SwiftUI `.labelReservedIconWidth(_:)` — reserves a minimum leading width for
/// a `Label`'s icon so titles stay aligned across rows that lack an icon.
///
/// LIMITATION: this API is only available on watchOS 26+ SDKs, newer than this
/// package's build floor (watchOS 9). Referencing the symbol — even behind
/// `#available` — would fail to compile on the supported build SDKs, so this is
/// a documented no-op: the view is returned unchanged. Registering the `$type`
/// keeps dispatch consistent and lets the no-op be swapped for a real
/// implementation once the SDK floor is raised.
enum RNWLabelReservedIconWidthModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("labelReservedIconWidth") { view, _, _ in view }
    }
}

File: src/modifiers/labelReservedIconWidth.ts
import { createModifier } from './createModifier';

/// SwiftUI `.labelReservedIconWidth(_:)`. Reserves a minimum leading width for
/// a `Label`'s icon so titles align even when some rows have no icon.
///
/// LIMITATION: `.labelReservedIconWidth(_:)` only exists on watchOS 26+ SDKs,
/// newer than this package's build floor. To keep the native side compiling on
/// every supported SDK it is currently a documented no-op — the view is
/// returned unchanged. The factory ships so callers can adopt it without churn
/// once the native side raises its SDK and wires it up.
export function labelReservedIconWidth(width: number) {
  return createModifier('labelReservedIconWidth', { width });
}
